integ-test/src/test/java/org/opensearch/sql/ppl/DedupCommandIT.java

```diff
@@ -62,4 +74,34 @@ public void testKeepEmptyDedup() throws IOException {
         rows("Virginia", null),
         rows("Dillard", 48086));
   }
+
+  private List<Object[]> extractActualRows(JSONObject result) {
+    JSONArray dataRows = result.getJSONArray("datarows");
+    List<Object[]> actualRows = new ArrayList<>();
+    for (int i = 0; i < dataRows.length(); i++) {
+      JSONArray row = dataRows.getJSONArray(i);
+      actualRows.add(new Object[] {row.get(0)});
+    }
+    return actualRows;
+  }
+
+  // Verify consecutive deduplication and create the expected rows
+  private List<Object[]> getExpectedDedupRows(List<Object[]> actualRows) {
```

Comment:
not quite sure about this..
is the data rows for dedup test case not being sorted causing failure in IT? because this seems like actual coding logic for dedup, instead of providing validated expected value (like the original `verifyDataRows(result, rows(true), rows(false), rows(true), rows(false))`

Reply:
No the failure was not because of the sorted value but the size of the data rows in the responses. Here we were hardcoding the expected response to be of size 4 but different data sources having different number of rows even though are de-duplicated, due to this hard coded check tests would fail. So instead of hard coding assertions, we can check the deduplication of the rows based on the response data.